src/System.Text.Json/tests/WritableJsonApiTests.cs

```diff
@@ -624,6 +966,324 @@ public static void TestAquiringAllProperties()
             };
 
             IEnumerable<JsonNode> fullTimeEmployees = employees.GetAllProperties("FTE");
+
+            Assert.Equal(3, fullTimeEmployees.Count());
+            Assert.True(fullTimeEmployees.Contains((JsonString)"John Smith"));
+            Assert.True(fullTimeEmployees.Contains((JsonString)"Ann Predictable"));
+            Assert.True(fullTimeEmployees.Contains((JsonString)"Byron Shadow"));
+        }
+
+        /// <summary>
+        /// Transforming JsoneNode to JsonElement
+        /// </summary>
+        [Fact]
+        public static void TestTransformingJsonNodeToJsonElement()
+        {
+            // Send Json through network, should not throw any exceptions:
+            JsonNode employeeDataToSend = EmployeesDatabase.GetNextEmployee().Value;
+            Mailbox.SendEmployeeData(employeeDataToSend.AsJsonElement());
+        }
+
+        /// <summary>
+        /// Transforming JsonElement to JsonNode
+        /// </summary>
+        [Fact]
+        public static void TestTransformingJsonElementToJsonNode()
+        {
+            // Retrieve Json from network, should not throw any exceptions:
+            JsonNode receivedEmployeeData = JsonElement.DeepCopy(Mailbox.RetrieveMutableEmployeeData());
+            if (receivedEmployeeData is JsonObject employee)
+            {
+                employee["name"] = new JsonString("Bob");
+                Mailbox.SendEmployeeData(employee.AsJsonElement());
+            }
+        }
+
+        /// <summary>
+        /// Transforming JsonDocument to JsonNode and vice versa
+        /// </summary>
+        [Fact]
+        public static void TestTransformingToFromJsonDocument()
+        {
+            string jsonString = @"
+            {
+                ""employee1"" : 
+                {
+                    ""name"" : ""Ann"",
+                    ""surname"" : ""Predictable"",
+                    ""age"" : 30,                
+                },
+                ""employee2"" : 
+                {
+                    ""name"" : ""Zoe"",
+                    ""surname"" : ""Coder"",
+                    ""age"" : 24,                
+                }
+            }";
+
+            using (JsonDocument employeesToSend = JsonDocument.Parse(jsonString))
+            {
+                // regular send:
+                Mailbox.SendAllEmployeesData(employeesToSend.RootElement);
+
+                // modified elements send:
+                JsonNode modifiableDocument = JsonElement.DeepCopy(employeesToSend);
+                var employees = modifiableDocument as JsonObject;
+                Assert.Equal(2, employees.Count());
+
+                employees.Add(EmployeesDatabase.GetNextEmployee());
```

Comment:
Just another thought on duplicates, it might be interesting to think about the case where `EmployeesDatabase.GetNextEmployee()` returns "employee1" as a key.